### ISSUE TITLE
v-if on portal component does not work with target-el

### ISSUE THREAD

**Issue by ParkerD559** [LinusBorg/portal-vue#163]
Demo: https://codesandbox.io/s/0y2wqy9w3w

I'm not sure if this is a bug (and may work after #139) or expected behavior because of limitations of the target being outside the vue app. In any case, I can't find any mention of a caveat with regards to using both `target-el`  and `v-if` in either [the doc section showing v-if being used](https://linusborg.github.io/portal-vue/#/guide?id=conditional-rendering-with-v-if) or [the doc section for target-el](https://linusborg.github.io/portal-vue/#/docs/portal?id=targetel).

**Comment by LinusBorg** [comment]
Yeah IK'm pretty sure that this one of the side effects of the (honestly) half-assed implementation I did with `targetEl`.

In 2.0.0, we will have a `<PortalTargetProvider>` component that will take care of cleanly mounting and dismounting Portals to external elements.

For 1.* this will be a wontfix, though I would consider merging a fix if someone felt the need to send a PR.

**Comment by LinusBorg** [comment]
I'm pretty much done with it, I developed it in a different repo and will move it into the `next` branch this week hopfully.

Beta will definitely be released before Christmas.

> I am having similar issues with this.$refs being unaccessible inside a portal on mounted() and looking for a solution.

Some details in a fresh issue would be helpful for me. From that description I can't tell you if 2.0.0 will solve your problem.

**Comment by morkro** [comment]
> Some details in a fresh issue would be helpful for me. From that description I can't tell you if 2.0.0 will solve your problem.

Fair enough. I've created a new ticket in #164, hope that helps. If it's basically a duplicate of #118, I am happy to continue the discussion there as well :)

**Comment by towry** [comment]
```js
<portal to="destination">
  <p>This slot content will be rendered wherever the <portal-target> with name 'destination'
    is  located.</p>
</portal>

<portal-target name="destination">
  <!--
  This component can be located anwhere in your App.
  The slot content of the above portal component will be rendered here.
  -->
</portal-target>
```

How can I use <portal-target> outside of Vue app?

**Comment by morkro** [comment]
@towry See here: https://linusborg.github.io/portal-vue/#/guide?id=rendering-outside-of-the-vue-app

Make sure to omit the `name=""` though, as this will not work. Here is an example:

**Vue template**
```html
<portal target-el="#dialog">
  <!-- ... -->
</portal>
```

**`index.html`**
```html
<body>
  <div id="app"></div>
  <div id="dialog"></div>
</body>
```

**Comment by towry** [comment]
@morkro I thought the `target-el` prop will be deprecated in next versions. I currently use this prop to render component outside of Vue app, directly mounted under the body element. The way I use it is to create a div dynamically and mount the component to it, and remove the div when my custom `portal-slot.vue` component `beforeDestroyed` being called. And days ago there is a bug from the production app that a popup modal wont close, I check it and turned out I use v-if on the `portal` but suddenly it doesn't work. While my custom `portal-slot.vue` component still work because I remove the div manually in the `beforeDestroyed` hook of my custom portal component. The version of `portal-vue` is `^1.1.1`.

**Comment by LinusBorg** [comment]
The targetEl  feature is redone for upcoming 2.0, i won't remove it but implementation and change. 

The new implementation won't have these sorts of problems.

**Comment by LinusBorg** [comment]
This should work fine in 2.0 with the new `<MountingPortal>` component, and is a `wont-fix` in 1.*, so I will close this.
